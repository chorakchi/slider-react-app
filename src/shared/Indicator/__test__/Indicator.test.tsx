
import React from "react";
import { shallow, mount } from "enzyme";
import * as renderer from "react-test-renderer";
import {Indicator} from "./../index";

describe("Indicator", () => {
  const defaultProps = {
    length: 3,
    onSelect: ()=> {},
    selected: 0,
  };

  it("should render the Indicator", () => {
    const subject = shallow(<Indicator {...defaultProps} />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Indicator {...defaultProps} />).toJSON();
    expect(subject).toMatchSnapshot();
  });
});