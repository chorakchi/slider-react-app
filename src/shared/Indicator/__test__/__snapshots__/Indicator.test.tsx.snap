// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator should render correctly and match to the Snapshot 1`] = `
<div
  className="sc-bdnylx ljOubv"
>
  <div
    className="sc-gtssRu iWqAEG"
    id="Indicator-0"
    onClick={[Function]}
    selected={true}
  />
  <div
    className="sc-gtssRu gomvNq"
    id="Indicator-1"
    onClick={[Function]}
    selected={false}
  />
  <div
    className="sc-gtssRu gomvNq"
    id="Indicator-2"
    onClick={[Function]}
    selected={false}
  />
</div>
`;
